spec/compose_spec.rb

```diff
@@ -0,0 +1,1094 @@
+        capture_stdout { described_class.process(args, "post" => true, "source" => "site") }
+        expect(path).to exist
+      end
+    end
+  end
+
+  context "posts with explicit --collection posts option" do
+    let(:name) { "A test post" }
+    let(:args) { [name] }
+    let(:options) { { "collection" => "posts" } }
+    let(:datestamp) { Time.now.strftime(Jekyll::Compose::DEFAULT_DATESTAMP_FORMAT) }
+    let(:timestamp) { Time.now.strftime(Jekyll::Compose::DEFAULT_TIMESTAMP_FORMAT) }
+    let(:filename) { "#{datestamp}-a-test-post.md" }
+    let(:path) { posts_dir.join(filename) }
+
+    before(:each) do
+      FileUtils.mkdir_p posts_dir unless File.directory? posts_dir
+      allow(Jekyll::Compose::FileEditor).to receive(:system)
+    end
+
+    after(:each) do
+      FileUtils.rm_r posts_dir if File.directory? posts_dir
+    end
+
+    it "creates a new post" do
+      expect(path).not_to exist
+      capture_stdout { described_class.process(args, "post" => true) }
+      expect(path).to exist
+    end
+
+    it "creates a post with a specified date" do
+      path = posts_dir.join "2012-03-04-a-test-post.md"
+      expect(path).not_to exist
+      capture_stdout { described_class.process(args, "post" => true, "date" => "2012-3-4") }
+      expect(path).to exist
+      expect(File.read(path)).to match(%r!date: 2012-03-04 00:00 \+0000!)
+    end
+
+    it "creates the post with a specified extension" do
+      html_path = posts_dir.join "#{datestamp}-a-test-post.html"
+      expect(html_path).not_to exist
+      capture_stdout { described_class.process(args, "post" => true, "extension" => "html") }
+      expect(html_path).to exist
+    end
+
+    it "creates a new post with the specified layout" do
+      capture_stdout { described_class.process(args, "post" => true, "layout" => "other-layout") }
+      expect(File.read(path)).to match(%r!layout: other-layout!)
+    end
+
+    it "should write a helpful message when successful" do
+      output = capture_stdout { described_class.process(args, "post" => true) }
+      expect(output).to include("New post created at #{File.join("_posts", filename).cyan}")
+    end
+
+    it "errors with no arguments" do
+      expect(lambda {
+        capture_stdout { described_class.process }
+      }).to raise_error("You must specify a name.")
+    end
+
+    it "errors with no arguments with post option" do
+      expect(lambda {
+        capture_stdout { described_class.process([], "post" => true) }
+      }).to raise_error("You must specify a name.")
+    end
+
+    it "creates the posts folder if necessary" do
+      FileUtils.rm_r posts_dir if File.directory? posts_dir
+      capture_stdout { described_class.process(args, "post" => true) }
+      expect(posts_dir).to exist
+    end
+
+    context "when the post already exists" do
+      let(:name) { "An existing post" }
+      let(:filename) { "#{Time.now.strftime(Jekyll::Compose::DEFAULT_DATESTAMP_FORMAT)}-an-existing-post.md" }
```

Comment:
Let us keep the line-length minimal:
```suggestion
      let(:filename) do
        "#{Time.now.strftime(Jekyll::Compose::DEFAULT_DATESTAMP_FORMAT)}-an-existing-post.md"
      end
```